projects/components/src/utils/dom-util.spec.ts

```diff
@@ -0,0 +1,139 @@
+            const el = this.widgetObject.getHiddenElement();
+            // Element should be hidden
+            expect(container.getBoundingClientRect().bottom).toBeLessThan(el.getBoundingClientRect().top);
+            DomUtil.scrollToElement(new ElementRef(el));
+            // It is a smooth scrolling, so we should give it a time
+            setTimeout(() => {
+                // Element should be visible
+                expect(container.getBoundingClientRect().bottom).toBeGreaterThan(el.getBoundingClientRect().top);
+                done();
+            }, 100);
+        });
+
+        it('does not scrolls the view if the element is already visible', function(this: Test, done): void {
+            const container = this.widgetObject.container;
+            const el = this.widgetObject.getVisibleElement();
+            const top = el.getBoundingClientRect().top;
+            // Element should be visible
+            expect(container.getBoundingClientRect().bottom).toBeGreaterThan(top);
+            DomUtil.scrollToElement(new ElementRef(container), TestHostComponent.getVisibleElementCss());
+            // It is a smooth scrolling, so we should give it a time
+            setTimeout(() => {
+                // Element should not have been scrolled at all
+                expect(el.getBoundingClientRect().top).toBe(top);
+                done();
+            }, 100);
+        });
+
+        it('does not scrolls the view if element defined by selector does not exist', function(this: Test, done): void {
+            const container = this.widgetObject.container;
+            const el = this.widgetObject.getHiddenElement();
+            const top = el.getBoundingClientRect().top;
+            DomUtil.scrollToElement(new ElementRef(container), '.not-existing');
+            // It is a smooth scrolling, so we should give it a time
+            setTimeout(() => {
+                expect(el.getBoundingClientRect().top).toBe(top);
+                done();
+            }, 100);
+        });
+    });
+});
+
+@Component({
+    template: `
+        <section>
+            <div class="container">
+                <div *ngFor="let el of elements; let i = index" class="element element-{{ i + 1 }}">{{ i + 1 }}</div>
+            </div>
+        </section>
+    `,
+    styles: [
+        `
+            .container {
+                height: 100px;
+                overflow: auto;
+            }
+            .element {
+                height: 20px;
+            }
+        `,
+    ],
+})
+export class TestHostComponent {
+    public static ELEMENTS_COUNT = 7;
+    public elements = Array(TestHostComponent.ELEMENTS_COUNT);
+    public static getHiddenElementCss(): string {
+        return `.element-${TestHostComponent.ELEMENTS_COUNT}`;
+    }
+    public static getVisibleElementCss(): string {
+        return `.element-3`;
+    }
+}
+
+export class TestHostWidgetObject extends WidgetObject<TestHostComponent> {
+    static tagName = `section`;
```

Comment:
👍 I always thought about this, we should use template strings everywhere instead of '' to discourage string concatenation